src/panels/lovelace/editor/delete-card.ts

```diff
@@ -16,8 +16,15 @@ export async function confDeleteCard(
     cardConfig,
     deleteCard: async () => {
       try {
-        await lovelace.saveConfig(deleteCard(lovelace.config, path));
-        showDeleteSuccessToast(element, hass!);
+        const newLovelace = deleteCard(lovelace.config, path);
+        await lovelace.saveConfig(newLovelace);
+        const action = async () => {
+          await lovelace.saveConfig(insertCard(newLovelace, path, cardConfig));
+        };
+        showDeleteSuccessToast(element, hass!, {
+          text: hass!.localize("ui.common.undo"),
```

Comment:
Now it's like why do we need the show delete toast helper, just the message is done for you. You could just keep with just setting the `undoAction` and set the text if the `undoAction` is defined in the helper.

Reply:
That was my original thought with the action. Its the showDelete Toast so why should I need to send in the undo action

Reply:
I get maybe there are times you don't want to undo? Maybe better to send in a boolean if you don't. 

```ts
showDeleteSuccessToast(element: HTMLElement, has: HomeAssistant, showUndo = true)
```

Reply:
Actually then you would need to send in Lovelace etc. I make the update 👍

Reply:
You need to send in the action if you want to show the button? Otherwise how would the toast know what to do? So if you pass an action (not the text) we should show the button.